### ISSUE TITLE
Drupal Not Installed Issue w/ Robo & blt tests command

### ISSUE THREAD

**Issue by mikemadison13** [acquia/blt#1478]
My system information:
* Operating system type: Mac OS
* Operating system version: 10.11.6
* BLT version: 8.x.dev 

Output of `blt doctor`:
```
$ blt doctor
Drupal VM was detected. Running blt doctor inside of VM...
'drush' cache was cleared.                                                                                             [success]
Changed current directory to /home/vagrant/.composer
+----------------------+--------------------------------------------------------------------------------+
| Property             | Value                                                                          |
+----------------------+--------------------------------------------------------------------------------+
| %paths               |                                                                                |
|  - %root             | /var/www/mtacc/docroot                                                         |
|  - %site             | sites/default                                                                  |
|  - %modules          | modules                                                                        |
|  - %themes           | themes                                                                         |
|  - %config-sync      | /var/www/mtacc/config/default                                                  |
|  - %files            | sites/default/files                                                            |
|  - %private          | /var/www/mtacc/files-private                                                   |
|  - %temp             | /tmp                                                                           |
| admin-theme          | seven                                                                          |
| blt-version          | 8.x-dev                                                                        |
| bootstrap            | Successful                                                                     |
| config-sync          | /var/www/mtacc/config/default                                                  |
| db-driver            | mysql                                                                          |
| db-hostname          | localhost                                                                      |
| db-name              | drupal                                                                         |
| db-password          | drupal                                                                         |
| db-port              | 3306                                                                           |
| db-status            | Connected                                                                      |
| db-username          | drupal                                                                         |
| drupal-settings-file | sites/default/settings.php                                                     |
| drupal-version       | 8.3.2                                                                          |
| drush-alias-files    | /var/www/mtacc/docroot/../drush/site-aliases/aliases.drushrc.php               |
|                      | /var/www/mtacc/docroot/../drush/site-aliases/example.acsf.aliases.drushrc.php  |
|                      | /var/www/mtacc/docroot/../drush/site-aliases/example.local.aliases.drushrc.php |
|                      | /var/www/mtacc/docroot/../drush/site-aliases/mtav2.aliases.drushrc.php         |
|                      | /var/www/mtacc/docroot/../drush/site-aliases/pcac.aliases.drushrc.php          |
| drush-conf           | /var/www/mtacc/drush/drushrc.php                                               |
| drush-script         | /var/www/mtacc/vendor/drush/drush/drush.php                                    |
| drush-temp           | /tmp                                                                           |
| drush-version        | 8.1.11                                                                         |
| files                | sites/default/files                                                            |
| install-profile      | lightning                                                                      |
| local-drushrc        | /var/www/mtacc/docroot/sites/default/local.drushrc.php                         |
| local-settings       | /var/www/mtacc/docroot/sites/default/settings/local.settings.php               |
| modules              | modules                                                                        |
| php-bin              | /usr/bin/php                                                                   |
| php-conf             | /etc/php/5.6/cli/php.ini                                                       |
| php-os               | Linux                                                                          |
| private              | /var/www/mtacc/files-private                                                   |
| root                 | /var/www/mtacc/docroot                                                         |
| site                 | sites/default                                                                  |
| temp                 | /tmp                                                                           |
| theme                | bartik                                                                         |
| themes               | themes                                                                         |
| uri                  | http://local.mtacc.com                                                         |
| user                 |                                                                                |
+----------------------+--------------------------------------------------------------------------------+
+--------------------------+---------------------------------------------------------------------------------+
| Check                    | Outcome                                                                         |
+--------------------------+---------------------------------------------------------------------------------+
| checkDocrootExists       | Found docroot.                                                                  |
| checkCoreExists          | Drupal core exists                                                              |
| checkSettingsFile        | BLT settings are included in settings file.                                     |
| checkLocalSettingsFile   | Found your local settings file.                                                 |
| checkLocalDrushFile      | Found your local drush settings file.                                           |
| checkUri                 | $options['uri'] is set.                                                         |
| checkUriResponse         | Received a response from site http://local.mtacc.com.                           |
| checkFileSystem:%files   | Public files directory is writable.                                             |
| checkFileSystem:%private | Private files directory is writable.                                            |
| checkFileSystem:%temp    | Temporary files directory is writable.                                          |
| checkDbConnection        | Connected to database.                                                          |
| checkDrupalBootstrapped  | Bootstrapped Drupal via drush.                                                  |
| checkDrupalInstalled     | Drupal is installed.                                                            |
| checkCaching:page        | Drupal cache is disabled.                                                       |
| checkCaching:css         | CSS preprocessing is disabled.                                                  |
| checkCaching:js          | JS preprocessing is disabled.                                                   |
| checkCiConfig            | Git remotes are set in project.yml.                                             |
| checkComposer:require    | acquia/blt is in composer.json's require object.                                |
| checkComposer:plugins    | hirak/prestissimo plugin for composer is installed.                             |
| checkBehat:exists        | Behat local settings file exists.                                               |
| checkProjectYml:keys     | project.yml has no deprecated keys.                                             |
| checkAcsfConfig          | BLT settings are not included in your pre-settings-php include.                 |
|                          |   Add a require statement for "/../vendor/acquia/blt/settings/blt.settings.php" |
|                          | to /var/www/mtacc/docroot/../factory-hooks/pre-settings-php/includes.php        |
| checkPhpDateTimezone     | PHP setting for date.timezone is correctly set                                  |
+--------------------------+---------------------------------------------------------------------------------+
[Exec] Done in 13.653s
```

When I run this command:
```
$ blt tests:security-updates
or
$ blt tests:behat
```

I get the following output:
```
[warning] Drupal is not installed.
Do you want to install Drupal? (y/n) 
```

And I expected this to happen:
I would expect the tests to run.

$ drush status shows me that the site is installed, blt doctor shows that the site is installed, and nothing is immediately jumping out at me in the BLT robo command for Behat that is even checking to see if Drupal is installed or not.

FWIW: if i actually run the drupal installation from this prompt, i still get the error

**Comment by grasmash** [comment]
I cannot reproduce this issue. Can you try the latest 8.x and verify that you're still experiencing it?

**Comment by mikemadison13** [comment]
this no longer seems to be an issue with the most current 8.x-dev. HOWEVER I am now getting: 
`$ blt tests:all
Executing command blt tests:all inside of Drupal VM...
> tests:behat
[error]  MySql is not available. Please run `blt doctor` to diagnose the issue. 
`

running doctor:
`$ blt doctor
Drupal VM was detected. Running blt doctor inside of VM...
Executing command cd /var/www/mtacc && /var/www/mtacc/vendor/bin/drush cc drush && /var/www/mtacc/vendor/bin/drush --include=/var/www/mtacc/vendor/acquia/blt/drush blt-doctor -r /var/www/mtacc/docroot inside of Drupal VM...
'drush' cache was cleared.                                                                                                             [success]
Changed current directory to /home/vagrant/.composer
+----------------------+--------------------------------------------------------------------------------+
| Property             | Value                                                                          |
+----------------------+--------------------------------------------------------------------------------+
| %paths               |                                                                                |
|  - %root             | /var/www/mtacc/docroot                                                         |
|  - %site             | sites/default                                                                  |
|  - %modules          | modules                                                                        |
|  - %themes           | themes                                                                         |
|  - %config-sync      | /var/www/mtacc/config/default                                                  |
|  - %files            | sites/default/files                                                            |
|  - %private          | /var/www/mtacc/files-private                                                   |
|  - %temp             | /tmp                                                                           |
| admin-theme          | seven                                                                          |
| blt-version          | 8.x-dev                                                                        |
| bootstrap            | Successful                                                                     |
| config-sync          | /var/www/mtacc/config/default                                                  |
| db-driver            | mysql                                                                          |
| db-hostname          | localhost                                                                      |
| db-name              | drupal                                                                         |
| db-password          | drupal                                                                         |
| db-port              | 3306                                                                           |
| db-status            | Connected                                                                      |
| db-username          | drupal                                                                         |
| drupal-settings-file | sites/default/settings.php                                                     |
| drupal-version       | 8.3.2                                                                          |
| drush-alias-files    | /var/www/mtacc/docroot/../drush/site-aliases/aliases.drushrc.php               |
|                      | /var/www/mtacc/docroot/../drush/site-aliases/example.acsf.aliases.drushrc.php  |
|                      | /var/www/mtacc/docroot/../drush/site-aliases/example.local.aliases.drushrc.php |
|                      | /var/www/mtacc/docroot/../drush/site-aliases/mtav2.aliases.drushrc.php         |
|                      | /var/www/mtacc/docroot/../drush/site-aliases/pcac.aliases.drushrc.php          |
| drush-conf           | /var/www/mtacc/drush/drushrc.php                                               |
| drush-script         | /var/www/mtacc/vendor/drush/drush/drush.php                                    |
| drush-temp           | /tmp                                                                           |
| drush-version        | 8.1.11                                                                         |
| files                | sites/default/files                                                            |
| install-profile      | lightning                                                                      |
| local-drushrc        | /var/www/mtacc/docroot/sites/default/local.drushrc.php                         |
| local-settings       | /var/www/mtacc/docroot/sites/default/settings/local.settings.php               |
| modules              | modules                                                                        |
| php-bin              | /usr/bin/php                                                                   |
| php-conf             | /etc/php/5.6/cli/php.ini                                                       |
| php-os               | Linux                                                                          |
| private              | /var/www/mtacc/files-private                                                   |
| root                 | /var/www/mtacc/docroot                                                         |
| site                 | sites/default                                                                  |
| temp                 | /tmp                                                                           |
| theme                | stark                                                                          |
| themes               | themes                                                                         |
| uri                  | http://local.mtacc.com                                                         |
| user                 |                                                                                |
+----------------------+--------------------------------------------------------------------------------+
+--------------------------+---------------------------------------------------------------------------------+
| Check                    | Outcome                                                                         |
+--------------------------+---------------------------------------------------------------------------------+
| checkDocrootExists       | Found docroot.                                                                  |
| checkCoreExists          | Drupal core exists                                                              |
| checkSettingsFile        | BLT settings are included in settings file.                                     |
| checkLocalSettingsFile   | Found your local settings file.                                                 |
| checkLocalDrushFile      | Found your local drush settings file.                                           |
| checkUri                 | $options['uri'] is set.                                                         |
| checkUriResponse         | Received a response from site http://local.mtacc.com.                           |
| checkFileSystem:%files   | Public files directory is writable.                                             |
| checkFileSystem:%private | Private files directory is writable.                                            |
| checkFileSystem:%temp    | Temporary files directory is writable.                                          |
| checkDbConnection        | Connected to database.                                                          |
| checkDrupalBootstrapped  | Bootstrapped Drupal via drush.                                                  |
| checkDrupalInstalled     | Drupal is installed.                                                            |
| checkCaching:page        | Drupal cache is disabled.                                                       |
| checkCaching:css         | CSS preprocessing is disabled.                                                  |
| checkCaching:js          | JS preprocessing is disabled.                                                   |
| checkCiConfig            | Git remotes are set in project.yml.                                             |
| checkComposer:require    | acquia/blt is in composer.json's require object.                                |
| checkComposer:plugins    | hirak/prestissimo plugin for composer is installed.                             |
| checkBehat:exists        | Behat local settings file exists.                                               |
| checkProjectYml:keys     | project.yml has no deprecated keys.                                             |
| checkAcsfConfig          | BLT settings are not included in your pre-settings-php include.                 |
|                          |   Add a require statement for "/../vendor/acquia/blt/settings/blt.settings.php" |
|                          | to /var/www/mtacc/docroot/../factory-hooks/pre-settings-php/includes.php        |
| checkPhpDateTimezone     | PHP setting for date.timezone is correctly set                                  |
+--------------------------+---------------------------------------------------------------------------------+`

**Comment by mikemadison13** [comment]
this seems to be a similar error to: https://github.com/acquia/blt/issues/1448 however it is occurring during the behat tests vs. the site install

**Comment by grasmash** [comment]
Resolved by #1495
